plugins/UsersManager/API.php

```diff
@@ -790,6 +844,16 @@ public function getTokenAuth($userLogin, $md5Password)
     {
         UsersManager::checkPasswordHash($md5Password, Piwik::translate('UsersManager_ExceptionPasswordMD5HashExpected'));
 
-        return md5($userLogin . $md5Password);
+        $user = $this->model->getUser($userLogin);
+
+        if (!$this->password->verify($md5Password, $user['password'])) {
+            return hash('sha256', $userLogin . microtime(true) . Common::generateUniqId());
```

Comment:
forget it, it's actually not needed in this case as it is only to keep BC etc. The regular logic is already in a method :)